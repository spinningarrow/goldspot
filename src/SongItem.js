import React from 'react'
import { playTrack } from './api'

const SongItem = ({ albumArt, title, artist, url, id }) => (
	<li className="song-item">
		<a
			href={url}
			target="_blank"
			rel="noopener noreferrer"
			onClick={playTrack(id)}
		>
			<img src={albumArt} alt="album art" />
			<span className="title">{title}</span>
		</a>
		<span className="artist">{artist}</span>
	</li>
)

export default SongItem
