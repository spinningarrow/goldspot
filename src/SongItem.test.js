import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import SongItem from './SongItem'

const renderer = new ShallowRenderer()

test('it displays the given details about the song item', () => {
	const tree = renderer
		.render(<SongItem albumArt="art.jpg" title="Dans L'sang" url="https://spotify/song" />)
	expect(tree).toMatchSnapshot()
})
