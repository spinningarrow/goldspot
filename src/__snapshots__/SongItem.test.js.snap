// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays the given details about the song item 1`] = `
<li
  className="song-item"
>
  <a
    href="https://spotify/song"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="album art"
      src="art.jpg"
    />
    <span
      className="title"
    >
      Dans L'sang
    </span>
  </a>
  <span
    className="artist"
  />
</li>
`;
